include/appbase/application_instance.hpp

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+namespace appbase {
+
+   using executor_t = appbase_executor;
+
+   // ------------------------------------------------------------------------------------------
+   class application : public application_base {
+   public:
+      static application&  instance() {
+         if (__builtin_expect(!!app_instance, 1))
+            return *app_instance;
+         app_instance.reset(new application);
+         return *app_instance;
+      }
+
+      static void reset_app_singleton() { app_instance.reset(); }
+      
+      static bool null_app_singleton()  { return !app_instance; }
+
+      template <typename Func>
+      auto post( int priority, Func&& func ) {
+         return application_base::post(executor_, priority, std::forward<Func>(func));
+      }
+
+      void exec() {
+         application_base::exec(executor_);
+      }
+
+      boost::asio::io_service& get_io_service() {
+         return executor_.get_io_service();
+      }
+
+      auto& get_priority_queue() {
+         return executor_.get_priority_queue();
+      }
```

Comment:
Seems like these are not needed. Just let users interact with `app().executor()`

Reply:
They are needed because they are used in `application_base`.

Reply:
OK, I have removed it.